docs/dev/background/ci.adoc

```diff
@@ -1,35 +1,102 @@
-* the typing is correct (with mypy)
-* there is no common vulnerability (with bandit and safety)
-* there is no license issue (with safety)
-* the unit tests run succesfully (with pytest)
-* the integration tests run succesfully (with pytest)
+== *Check code quality* workflow
 
-It also generates and publish a sonarqube report on code quality.
+This workflow is run either automatically after each `git push` command on each branch and pull request, or manually.
 
-== generate documentation workflow
+It runs the following jobs:
 
-This workflow build the documentation using the following command :
-[,yml]
+* Check format (pre-commit, black, isort)
+    ** This job checks that you have run `pre-commit run --all-files` in your local git repository before committing. The `pre-commit` runs `black` and `isort` to auto-format your source code and facilitate reviewing differences and merging between git branches.
+
+* Check linting (pylint, flake8)
+    ** These linters analyse your code without actually running it. They check for errors, enforce a coding standard, look for code smells, and can make suggestions about how the code could be refactored.
+
+* Check typing (mypy)
+    ** Add type annotations to your Python programs, and use mypy to type check them.
+
+* Check security (bandit, safety, trivy)
+    ** Bandit is a tool designed to find common security issues in Python code.
+    ** Safety checks Python dependencies for known security vulnerabilities and suggests the proper remediations for vulnerabilities detected.
+    ** Trivy is a comprehensive and versatile security scanner that looks for security issues, and targets where it can find those issues. The scan results are updated into the GitHub Security tab: https://github.com/RS-PYTHON/rs-server/security/code-scanning
+
+* Run unit tests (pytest)
+* Run integration tests (pytest)
+    ** Test source code and determine if it is fit to use.
+    ** Unit tests are faster than integration tests and use mockup servers. Integration tests are run on the production infrastructure (to be confirmed).
+    ** They also calculate code coverage i.e. percentage measure of the degree to which the source code is executed when the tests are run. Only the unit tests coverage is shown in sonarqube.
+
+* Quality report (sonarqube)
+    ** SonarQube is a platform for continuous inspection of code quality to perform automatic reviews with static analysis of code to detect bugs and code smells. It offers reports on duplicated code, coding standards, unit tests, code coverage, code complexity, comments, bugs, and security recommendations.
+    ** It renders the reports from pylint, flake8, bandit and code coverage in a more graphical way.
+    ** Reports are displayed at: https://sonarqube.ops-csc.com/dashboard?branch=develop&id=RS-PYTHON_rs-server_AYw0m7ixvQv-JMsowILQ
+
+== *Publish wheels and Docker images* workflow
+
+This workflow is run either automatically after adding a git tag to a branch, or manually. It:
```

Comment:
This workflow is run either automatically after adding a git tag to a commit, or manually. It: